app/models/orchestration_stack.rb

```diff
@@ -16,9 +16,17 @@ class OrchestrationStack < ActiveRecord::Base
   belongs_to :orchestration_template
   belongs_to :cloud_tenant
 
-  has_many   :vms, :class_name => "ManageIQ::Providers::CloudManager::Vm"
-  has_many   :security_groups
-  has_many   :cloud_networks
+  has_many   :direct_vms,             :class_name => "ManageIQ::Providers::CloudManager::Vm"
+  has_many   :direct_security_groups, :class_name => "SecurityGroup"
+  has_many   :direct_cloud_networks,  :class_name => "CloudNetwork"
+
+  virtual_has_many :all_vms, :class_name => "ManageIQ::Providers::CloudManager::Vm"
+  virtual_has_many :all_security_groups
+  virtual_has_many :all_cloud_networks
```

Comment:
@bzwei There is no need to continue the duplication of the "all_*" methods.  Initially the "all_vms" was added to the service model before the "vms" relationship existed.  Later it was decided that "vms" would represent VMs defined for the service and any sub-services.  We should look into deprecating "all_vms" in the service model but that does not have to be in this PR.